.github/workflows/nextjs-integration-test.yml

```diff
@@ -0,0 +1,158 @@
+        with:
+          targets: x86_64-unknown-linux-musl
+          cache-key: dev-x86_64-unknown-linux-musl
+          save-cache: true
+
+      - name: Install musl tools
+        run: |
+          wget https://github.com/napi-rs/napi-rs/releases/download/linux-musl-cross%4011.2.1/x86_64-linux-musl-native.tgz -O musl.tgz
+          tar -xvzf musl.tgz
+          sudo mv x86_64-linux-musl-native /usr/x86_64-linux-musl
+          sudo ln -sf /usr/x86_64-linux-musl/bin/x86_64-linux-musl-cc /usr/bin/musl-gcc
+          sudo ln -sf /usr/x86_64-linux-musl/bin/x86_64-linux-musl-g++ /usr/bin/musl-g++
+
+      - name: Build debug next-dev (rustls-tls)
+        run: |
+          cargo build -p next-dev --target x86_64-unknown-linux-musl --no-default-features --features cli,custom_allocator,rustls-tls
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: debug-next-dev-linux-musl
+          path: |
+            target/x86_64-unknown-linux-musl/debug/next-dev
+
+  # Run actual next.js integration test
+  execute_tests:
+    needs: [rust_build_dev]
+    # This job name is being used in github action to collect test results. Do not change it, or should update
+    # ./.github/actions/next-integration-test to match the new name.
+    name: Next.js integration test
+    runs-on: ubuntu-latest-8-core-oss
+    strategy:
+      fail-fast: false
+      matrix:
+        node: [16, 18]
+        group: [1, 2]
+
+    env:
+      # Enabling backtrace will makes snapshot tests fail
+      RUST_BACKTRACE: 0
+      NEXT_TELEMETRY_DISABLED: 1
+      # Path to the next-dev binary located in **docker container** image.
+      NEXT_DEV_BIN: /work/next-dev
+      # Glob pattern to run specific tests with --turbo.
+      NEXT_DEV_TEST_GLOB: "*"
+      # pnpm version should match to what upstream next.js uses
+      PNPM_VERSION: 7.24.3
+
+    steps:
+      - name: Find Next.js latest release version
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          # Grab the latest release version from next.js repo, including prelease. `/releases/latest` will only return latest stable release.
+          echo NEXJS_LATEST_VERSION=$(gh release --repo vercel/next.js --limit 1 list | sed -n 1p | awk '{print $1}') >> $GITHUB_ENV
+      - name: Set Next.js release version
+        run: |
+          echo "NEXTJS_VERSION=${{ inputs.version != '' && inputs.version || env.NEXJS_LATEST_VERSION }}" >> $GITHUB_ENV
+          echo "Checking out Next.js ${{ env.NEXTJS_VERSION }}"
+
+      # https://github.com/actions/virtual-environments/issues/1187
+      - name: tune linux network
+        run: sudo ethtool -K eth0 tx off rx off
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          repository: vercel/next.js
+          ref: ${{ env.NEXTJS_VERSION }}
+
+      - uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: ./*
+          key: ${{ github.sha }}-${{ github.run_number }}
```

Comment:
What exactly is this supposed to do?

I think as is, this will cache the entire repository and never restore anything

Reply:
it's probably copy-paste from next.js upstream, I'll try to double check later.